File: app/src/main/java/com/sharpdroid/registroelettronico/database/room/dao/NoteDao.java
package com.sharpdroid.registroelettronico.database.room.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.sharpdroid.registroelettronico.database.entities.Note;

import java.util.List;

@Dao
public interface NoteDao {

    @Query("SELECT * FROM NOTE WHERE PROFILE=:profile ORDER BY M_DATE DESC")
    LiveData<List<Note>> getNotes(long profile);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<Note> notes);

    @Query("DELETE FROM NOTE WHERE PROFILE = :profile")
    void delete(long profile);
}

File: app/src/main/java/com/sharpdroid/registroelettronico/database/dao/NoteDao.kt
package com.sharpdroid.registroelettronico.database.dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

import com.sharpdroid.registroelettronico.database.entities.Note

@Dao
interface NoteDao {

    @Query("SELECT * FROM NOTE WHERE PROFILE=:profile ORDER BY M_DATE DESC")
    fun getNotes(profile: Long): LiveData<List<Note>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(notes: List<Note>)

    @Query("DELETE FROM NOTE WHERE PROFILE = :profile")
    fun delete(profile: Long)
}
